pkg/controllers/hub/trafficmanagerprofile/controller.go

```diff
@@ -313,6 +315,37 @@ func generateAzureTrafficManagerProfile(profile *fleetnetv1beta1.TrafficManagerP
 	}
 }
 
+// buildAzureTrafficManagerProfileRequest assumes desired is always valid.
+func buildAzureTrafficManagerProfileRequest(current, desired armtrafficmanager.Profile) armtrafficmanager.Profile {
```

Comment:
I wonder if we want to return a "changed" boolean and only call createdOrUpdate when anything changed is true. Cloud provider does that.

Reply:
if there is no change (checked in 148), it will return directly in line 151.